src/middlewared/middlewared/plugins/smb.py

```diff
@@ -584,104 +589,43 @@ class Config:
         datastore_extend = 'sharing.smb.extend'
         cli_namespace = 'sharing.smb'
         role_prefix = 'SHARING_SMB'
+        entry = SmbShareEntry
 
-    @accepts(
-        Dict(
-            'sharingsmb_create',
-            Str('purpose', enum=[x.name for x in SMBSharePreset], default=SMBSharePreset.DEFAULT_SHARE.name),
-            SchemaPath('path', required=True),
-            Str('path_suffix', default=''),
-            Bool('home', default=False),
-            Str('name', max_length=80, required=True),
-            Str('comment', default=''),
-            Bool('ro', default=False),
-            Bool('browsable', default=True),
-            Bool('timemachine', default=False),
-            Int('timemachine_quota', default=0),
-            Bool('recyclebin', default=False),
-            Bool('guestok', default=False),
-            Bool('abe', default=False),
-            List('hostsallow'),
-            List('hostsdeny'),
-            Bool('aapl_name_mangling', default=False),
-            Bool('acl', default=True),
-            Bool('durablehandle', default=True),
-            Bool('shadowcopy', default=True),
-            Bool('streams', default=True),
-            Bool('fsrvp', default=False),
-            Str('auxsmbconf', max_length=None, default=''),
-            Bool('enabled', default=True),
-            Bool('afp', default=False),
-            Dict(
-                'audit',
-                Bool('enable'),
-                List('watch_list', default=NOT_PROVIDED),
-                List('ignore_list', default=NOT_PROVIDED)
-            ),
-            register=True
-        ),
-        audit='SMB share create', audit_extended=lambda data: data['name']
+    @api_method(
+        SmbShareCreateArgs, SmbShareCreateResult,
+        audit='SMB share create',
+        audit_extended=lambda data: data['name']
     )
     @pass_app(rest=True)
```

Comment:
Last thing, just letting you know we have a way to combine these decorators now.
```suggestion
    @api_method(
        SmbShareCreateArgs, SmbShareCreateResult,
        audit='SMB share create',
        audit_extended=lambda data: data['name'],
        pass_app=True,
        pass_app_rest=True,
    )
```